axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -1667,4 +1700,102 @@ public BigDecimal adjustAmountInCompanyCurrency(
   public boolean isPartiallyPaid(InvoiceTerm invoiceTerm) {
     return invoiceTerm.getAmount().compareTo(invoiceTerm.getAmountRemaining()) != 0;
   }
+
+  @Override
+  public InvoiceTerm initInvoiceTermWithParents(InvoiceTerm invoiceTerm) throws AxelorException {
+    Invoice invoice = invoiceTerm.getInvoice();
+    MoveLine moveLine = invoiceTerm.getMoveLine();
+
+    if (invoice == null && moveLine != null && moveLine.getMove() != null) {
+      this.initCustomizedInvoiceTerm(moveLine, invoiceTerm, moveLine.getMove());
+    } else if (invoice != null) {
+      this.initCustomizedInvoiceTerm(invoice, invoiceTerm);
+    }
+
+    setParentFields(invoiceTerm, moveLine != null ? moveLine.getMove() : null, moveLine, invoice);
+    return invoiceTerm;
+  }
+
+  @Override
+  public boolean setShowFinancialDiscount(InvoiceTerm invoiceTerm) {
+    Invoice invoice = invoiceTerm.getInvoice();
+    MoveLine moveLine = invoiceTerm.getMoveLine();
+    AppAccount appAccount = appAccountService.getAppAccount();
+    if (appAccount == null || !appAccount.getManageFinancialDiscount()) {
+      return false;
+    }
+
+    if (invoice != null) {
+      return new ArrayList<>(
+              Arrays.asList(InvoiceRepository.STATUS_VENTILATED, InvoiceRepository.STATUS_CANCELED))
+          .contains(invoice.getStatusSelect());
+    }
+    if (moveLine != null && moveLine.getMove() != null) {
+      return new ArrayList<>(
+              Arrays.asList(
+                  MoveRepository.STATUS_DAYBOOK,
+                  MoveRepository.STATUS_ACCOUNTED,
+                  MoveRepository.STATUS_CANCELED))
+          .contains(moveLine.getMove().getStatusSelect());
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isPaymentConditionFree(InvoiceTerm invoiceTerm) {
+    if (invoiceTerm.getInvoice() != null) {
+      return Optional.of(invoiceTerm.getInvoice())
+          .map(Invoice::getPaymentCondition)
+          .map(PaymentCondition::getIsFree)
+          .orElse(false);
+    } else if (invoiceTerm.getMoveLine() != null) {
+      return Optional.of(invoiceTerm.getMoveLine())
+          .map(MoveLine::getMove)
+          .map(Move::getPaymentCondition)
+          .map(PaymentCondition::getIsFree)
+          .orElse(false);
+    }
+
+    return false;
+  }
+
+  @Override
+  public List<DMSFile> getLinkedDmsFile(InvoiceTerm invoiceTerm) {
+    Move move =
+        Optional.of(invoiceTerm).map(InvoiceTerm::getMoveLine).map(MoveLine::getMove).orElse(null);
+    if (move != null) {
+      return DMSFileRepo.all()
+          .filter(
+              String.format(
+                  "self.isDirectory = false AND self.relatedId = %d AND self.relatedModel = 'com.axelor.apps.account.db.Move'",
+                  move.getId()))
+          .fetch();
+    }
+    return new ArrayList<>();
```

Comment:
Please make this return first if `move == null` and then the other return outside of the condition block